### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by tyler-reitz**
Ok, let me know what you think. There might be some overkill here, like testing each component's props for children, styles, className, etc. I mainly focused on testing the prop-types and the SC rendered styles. This is my first go at testing components that are mostly just UI, so feedback is welcome!

### CODE REVIEW

File: src/components/Anchor/Anchor.test.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  it("warns if props are missing or the wrong type", () => {
+    expect(() => { assertPropTypes(Anchor.propTypes, { ...testProps, children: undefined}, 'prop', Anchor.children); }).toThrow();
+    expect(() => { assertPropTypes(Anchor.propTypes, { ...testProps, href: undefined }, 'prop', Anchor.href); }).toThrow();
+    expect(() => { assertPropTypes(Anchor.propTypes, { ...testProps, className: 123 }, 'prop', Anchor.className); }).toThrow();
+    expect(() => { assertPropTypes(Anchor.propTypes, { ...testProps, style: "red" }, 'prop', Anchor.style); }).toThrow();
+  });
+
+  it("renders", () => {
+    const wrapper = shallow(<Anchor />);
+    expect(wrapper.find("Anchor__StyledAnchor").length).toBe(1);
```
Comment (maracuja-juice) [inline src/components/Anchor/Anchor.test.js]: Can't you also do:
wrapper.find(Anchor) ?
How many elements are even in the wrapper? I feel like you can also just do: expect(wrapper).toHaveLengt(1)

File: src/components/AppBar/AppBar.test.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  });
+
+  it("renders default props", () => {
+    const wrapper = shallow(<AppBar />);
+    expect(wrapper.props().shadow).toBe(true);
+    expect(wrapper.props().fixed).toBe(true);
+  });
+
+  it("renders default component styles", () => {
+    const tree = renderer.create(<AppBar theme={themes.default} />).toJSON();
```
Comment (maracuja-juice) [inline src/components/AppBar/AppBar.test.js]: Not sure if these kind of tests are valuable

File: src/components/common/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { css } from "styled-components";
 
-export const shadow = `4px 4px 10px 0 rgba(0, 0, 0, 0.35)`;
+export const shadow = `4px 4px 10px 0 rgba(0,0,0,0.35)`;
 export const insetShadow = `inset 3px 3px 10px rgba(0, 0, 0, 0.2)`;
```
Comment (maracuja-juice) [inline src/components/common/index.js]: I think you should revert these formatting changes

File: src/components/AppBar/AppBar.test.js (hunk 2 above)
Comment (goodmind) [reply src/components/AppBar/AppBar.test.js]: Snapshot tests aren't valuable at all, better do some screenshot testing

Comment (maracuja-juice) [reply src/components/AppBar/AppBar.test.js]: I was referring to the highlighted test though: "renders default component styles"
I do personally find snapshot tests valuable.

File: src/components/Anchor/Anchor.test.js (hunk 1 above)
Comment (tyler-reitz) [reply src/components/Anchor/Anchor.test.js]: expect(wrapper).toHaveLength(1) is cleaner for sure.

File: src/components/AppBar/AppBar.test.js (hunk 2 above)
Comment (tyler-reitz) [reply src/components/AppBar/AppBar.test.js]: The readme for jest-styled-components had it as an [example](https://github.com/styled-components/jest-styled-components#tohavestylerule) in their docs, so thought I'd follow suit. Not sure how valuable they are either tbh. Maybe just testing the interpolated styles is enough? I'd be happy to add some screenshot tests as well.

File: src/components/common/index.js (hunk 3 above)
Comment (tyler-reitz) [reply src/components/common/index.js]: Sure thing. They were made in response to the following test failure I kept getting. 

    Component: AppBar › renders default component styles

    "Value mismatch for property 'box-shadow'"

    Expected
      "box-shadow: 4px 4px 10px 0 rgba(0, 0, 0, 0.35),inset 1px 1px 0px 1px #dfe0e3,inset -1px -1px 0 1px #888c8f"
    Received:
      "box-shadow: 4px 4px 10px 0 rgba(0,0,0,0.35),inset 1px 1px 0px 1px #dfe0e3,inset -1px -1px 0 1px #888c8f"

But if they aren't that valuable I'll go ahead revert them.

File: src/components/Anchor/Anchor.test.js (hunk 1 above)
Comment (arturbien) [reply src/components/Anchor/Anchor.test.js]: Yeah, @maracuja-juice is right. Can you fix that Tyler? 😊

Comment (tyler-reitz) [reply src/components/Anchor/Anchor.test.js]: Sure thing, will do.

File: src/components/AppBar/AppBar.test.js (hunk 2 above)
Comment (arturbien) [reply src/components/AppBar/AppBar.test.js]: Tbh I don't know if we should keep this kind of tests. (I will have to do some research 😂). Lets leave it like it is for now and we will get back to this later

File: src/components/common/index.js (hunk 3 above)
Comment (arturbien) [reply src/components/common/index.js]: We can keep that. We will have to change the insetShadow later on too if we wanna test for styles (text inputs and Cutout uses this variable)

**Comment by tyler-reitz** [comment]
Ok, feedback has been incorporated.

**Pull request merged**